### PR TITLE
feat: P4ADEV-1632 classify-iuv-activity-retrieve-iuf

### PR DESCRIPTION
**Pull request by silvanotamburini**
#### Description

ClassifyIuvActivity - retrieve IUF

#### Motivation and Context

Activities

#### How Has This Been Tested?

- Pre-Deploy Test
    - [x] Unit
    - [ ] Integration (Narrow)
- Post-Deploy Test
    - [ ] Isolated Microservice
    - [ ] Broader Integration
    - [ ] Acceptance
    - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [ ] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/TransferClassificationActivity.java [hunk 1]
```diff
@@ -19,4 +22,16 @@ public interface TransferClassificationActivity {
...
+	 * retrieve Transfer data by semantic key
+	 *
+	 * @param orgId the unique identifier of the organization
+	 * @param iuv   the unique identifier of the payment (IUV)
+	 * @param iur   the identifier of the receipt associated with the payment
+	 * @param transferIndex  the index of the transfer to be classified
+	 * @return List of PaymentsReportingDTO object containing data found (may be 1 object)
+	 */
+	@ActivityMethod
+	List<PaymentsReportingDTO> retrievePaymentReportingBySemanticKey(Long orgId, String iuv, String iur, int transferIndex);
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/TransferClassificationActivity.java:36]: you should not add a new method on the activity, you should just call the findBySemanticKey of PaymentsReportingDao inside the `classify` method

File: src/main/java/it/gov/pagopa/payhub/activities/dao/PaymentsReportingDao.java [hunk 2]
```diff
@@ -18,4 +18,15 @@ public interface PaymentsReportingDao {
...
+	/**
+	 * find payment reporting by semantic key
+	 *
+	 * @param orgId  organization id
+	 * @param iuv    payment identifier
+	 * @param iur    reporting identifier
+	 * @param transferIndex transfer index
+	 * @return  List of objects PaymentsReportingDTO returned (may be 1 object)
+	 */
+	List<PaymentsReportingDTO> findBySemanticKey(Long orgId, String iuv, String iur, int transferIndex);
```
Comment (antonioT90) [inline src/main/java/it/gov/pagopa/payhub/activities/dao/PaymentsReportingDao.java:31]: it should find at most one record, not a List
```suggestion
	PaymentsReportingDTO findBySemanticKey(Long orgId, String iuv, String iur, int transferIndex);
```

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/TransferClassificationActivity.java (hunk 1 above)
Comment (silvanotamburini) [reply src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/TransferClassificationActivity.java:36]: OK

File: src/main/java/it/gov/pagopa/payhub/activities/dao/PaymentsReportingDao.java (hunk 2 above)
Comment (silvanotamburini) [reply src/main/java/it/gov/pagopa/payhub/activities/dao/PaymentsReportingDao.java:31]: OK

**Review by antonioT90 — APPROVED** [review]

**Pull request merged**